packages/opentelemetry-core/src/trace/GlobalTracerDelegate.ts

```diff
@@ -17,18 +17,35 @@
 import * as types from '@opentelemetry/types';
 import { NoopTracer } from './NoopTracer';
 
-const noopTracer = new NoopTracer();
-let _globalTracer: types.Tracer | null = null;
-
+// Acts a bridge to the global tracer that can be safely called before the
+// global tracer is initialized. The purpose of the delegation is to avoid the
+// sometimes nearly intractible initialization order problems that can arise in
+// applications with a complex set of dependencies. Also allows for the tracer
+// to be changed/disabled during runtime without needing to change reference
+// to the global tracer
 export class GlobalTracerDelegate implements types.Tracer {
+  private _tracer: types.Tracer | null;
+  private readonly _fallbackTracer: types.Tracer;
+
+  // Wrap a tracer with a GlobalTracerDelegate. Provided tracer becomes the default
+  // fallback tracer for when a global tracer has not been initialized
+  constructor(tracer?: types.Tracer | null, fallbackTracer?: types.Tracer) {
+    this._tracer = tracer || null;
+    this._fallbackTracer = fallbackTracer || new NoopTracer();
+  }
+
+  set tracer(tracer: types.Tracer | null) {
```

Comment:
What would you think about not giving the ability to get/set the tracer and just relying on the constructor? That way if customers want different sorts of tracers, they could create new instances with different arguments.

What if instead we exposed a `start`  functions on the `Tracer` interface, and made a `StoppableTracer` interface that adds a `stop` method that TracerDelegate implements by swapping out the tracer for the fallback tracer.

Reply:
Big +1 for not allowing to change the tracer at runtime. That sounds like a footgun with no added value.

Reply:
Can users ever use the underlying tracer directly? It would be much easier to simply add `start` and `stop` if you _have_ to use a delegate.

Reply:
Added `start`/`stop` to the delegate in 709670d